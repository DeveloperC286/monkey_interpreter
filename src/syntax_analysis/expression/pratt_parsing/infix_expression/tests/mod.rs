use rstest::rstest;

#[rstest(
    code,
    snapshot_name,
    case("10 > 5 == true", "test_syntax_analysis_for_infix_expression_case8"),
    case("10 < 5 != false", "test_syntax_analysis_for_infix_expression_case9")
)]
fn test_syntax_analysis_for_infix_expression(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name);
}
