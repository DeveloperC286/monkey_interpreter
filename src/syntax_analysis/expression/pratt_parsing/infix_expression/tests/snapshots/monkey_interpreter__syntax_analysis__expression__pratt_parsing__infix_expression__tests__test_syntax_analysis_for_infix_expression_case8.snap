---
source: src/syntax_analysis/expression/pratt_parsing/infix_expression/tests/mod.rs
expression: "crate::syntax_analysis::get_abstract_syntax_tree(crate::lexical_analysis::LexicalAnalysis::from(code))"

---
AbstractSyntaxTree {
    abstract_syntax_tree: [
        Expression {
            expression: Infix {
                left_hand: Infix {
                    left_hand: Integer {
                        integer_token: Integer {
                            literal: "10",
                        },
                    },
                    operator_token: GreaterThan,
                    right_hand: Integer {
                        integer_token: Integer {
                            literal: "5",
                        },
                    },
                },
                operator_token: Equals,
                right_hand: Boolean {
                    boolean_token: True,
                },
            },
        },
    ],
    syntax_parsing_errors: [],
}
